packages/core/feature-flags/core-feature-flags-browser/src/types.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Provider } from '@openfeature/web-sdk';
+import { type EvaluationContext as OpenFeatureEvaluationContext } from '@openfeature/core';
+import type { Observable } from 'rxjs';
+
+/**
+ * The evaluation context to use when retrieving the flags.
+ * @public
+ */
+export type EvaluationContext = OpenFeatureEvaluationContext & {
+  /**
+   * We use multi-context so that we can apply segmentation rules at different levels (`organization`/`kibana`).
+   * * `organization` includes any information that it's common to all the projects/deployments in an organization. An example is the in_trial status.
+   * * The `kibana` context includes all the information that identifies a project/deployment. Examples are version, offering, and has_data.
+   * Kind helps us specify which sub-context should receive the new attributes.
+   * If no `kind` is provided, it defaults to `kibana`.
+   * @public
+   *
+   * @example Providing properties for both contexts
+   * {
+   *   kind: 'multi',
+   *   organization: {
+   *     key: 1234,
+   *     in_trial: true,
+   *   },
+   *   kibana: {
+   *     key: 12345567890,
+   *     version: 8.15.0,
+   *     buildHash: 'ffffffffaaaaaaaa',
+   *   },
+   * }
+   *
+   * @example Appending context to the organization sub-context
+   * {
+   *   kind: 'organization',
+   *   key: 1234,
+   *   in_trial: true,
+   * }
+   *
+   * @example Appending context to the `kibana` sub-context
+   * {
+   *   key: 12345567890,
+   *   version: 8.15.0,
+   *   buildHash: 'ffffffffaaaaaaaa',
+   *   }
+   * }
+   */
+  kind?: 'multi' | 'organization' | 'kibana';
```

Comment:
Non-blocking: I was trying to figure out if we could use TS to enforce this more cleanly. But the `Record<string...>` part does not really allow for TS to guide developers more.

I came up with this, but it's a breaking change.

```typescript
export type EvaluationContext =
  | {
      kind: 'multi';
      values: {
        organization?: OpenFeatureEvaluationContext;
        kibana?: OpenFeatureEvaluationContext;
      };
    }
  | {
      // defaults to 'kibana'
      kind?: 'kibana' | 'organization';
      values: OpenFeatureEvaluationContext;
    };

```

Reply:
I think I found a way: [`5881d9e` (#188562)](https://github.com/elastic/kibana/pull/188562/commits/5881d9e9c852ee87ff3aa33a8ea16eef59ca58f1)

Reply:
That's definitely a nicer declaration. I was just playing around with it:

```
const e: EvaluationContext = { /* press ctrl + space */ };
```

And trying to get it to force me to end up at one of your examples. Right now TS will accept `{ kibana: {}, organization: {} }` (without the `kind: 'multi'` part and default to `kind: 'kibana'`).

Not major, I think your examples cover the cases quite nicely.